s3/pkg/service/object.go

```diff
@@ -164,6 +165,18 @@ func (s *s3Service) PutObject(ctx context.Context, in pb.S3_PutObjectStream) err
 		log.Errorln("failed to get backend client with err:", err)
 		return err
 	}
+	// TODO Prakash(rhsakarpos@gmail.com) once Neelam checks in the GET versioning on bucket code, add it here
+	isBucketVersioned := true
+	if isBucketVersioned{
+		versionId, now := utils.GetVersionId()
+		obj.VersionId = versionId
+		obj.LastModified = now
+		// update the cloud backend object name here
+		obj.ObjectKey = obj.ObjectKey + "_" + strconv.FormatUint(obj.VersionId,10)
```

Comment:
Does this will conclude a different object key in database? Actually we have obj.objectId which is used to store the key used in cloud backend.

Reply:
Hi @rhsakarpos please check this, cause different object key means a different object, I guess this is not supposed to be here, is it?

Reply:
yes, different version = different object, sorry, did not get the problem...

Reply:
For example, user put object twice which use obj as it's key, and then user will have two versions of the same key 'obj', not obj_xxxxxx and obj_yyyyy, right?